jobs/cjd_sample/src/main/java/com/google/samples/CompanyAndJobCrudSample.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples;
+
+import com.google.api.services.jobs.v2.JobService;
+import com.google.api.services.jobs.v2.model.Company;
+import com.google.api.services.jobs.v2.model.CreateJobRequest;
+import com.google.api.services.jobs.v2.model.CustomAttribute;
+import com.google.api.services.jobs.v2.model.Job;
+import com.google.api.services.jobs.v2.model.StringValues;
+import com.google.api.services.jobs.v2.model.UpdateJobRequest;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Company and Job CRUD samples.
+ */
+public final class CompanyAndJobCrudSample {
+
+  private static JobService jobService = JobServiceUtils.getJobService();
+
+  // [START basic_company]
+
+  /**
+   * Generate a company
+   */
+  public static Company generateCompany() throws IOException {
+    // distributor company id should be a unique Id in your system.
+    String distributorCompanyId = "company:" + String.valueOf(new Random().nextLong());
+
+    Company company =
+        new Company()
+            .setDisplayName("Google")
+            .setHqLocation("1600 Amphitheatre Parkway Mountain View, CA 94043")
+            .setDistributorCompanyId(distributorCompanyId);
+    System.out.println("Company generated: " + company);
+    return company;
+  }
+  // [END basic_company]
+
+  // [START create_company]
+
+  /**
+   * Create a company.
+   */
+  public static Company createCompany(Company companyToBeCreated) throws IOException {
+    Company companyCreated = jobService.companies().create(companyToBeCreated).execute();
+    System.out.println("Company created: " + companyCreated);
+    return companyCreated;
+  }
+  // [END create_company]
+
+  // [START get_company]
+
+  /**
+   * Get a company.
+   */
+  public static Company getCompany(String companyName) throws IOException {
+    Company companyExisted = jobService.companies().get(companyName).execute();
+    System.out.println("Company existed: " + companyExisted);
```

Comment:
What if the company doesn't exist?

Reply:
It would receive a GoogleJsonResponseException. That's the reason we made the CRUD together.